2.0/m_antirandom.cpp

```diff
@@ -564,52 +582,27 @@ class ModuleAntiRandom : public Module
 			}
 
 			/* Check consonants */
-			switch (*s)
+			if (strchr("bcdfghjklmnpqrstvwxz", c))
 			{
-				case 'b':
-				case 'c':
-				case 'd':
-				case 'f':
-				case 'g':
-				case 'h':
-				case 'j':
-				case 'k':
-				case 'l':
-				case 'm':
-				case 'n':
-				case 'p':
-				case 'q':
-				case 'r':
-				case 's':
-				case 't':
-				case 'v':
-				case 'w':
-				case 'x':
-				case 'z':
-					consonants++;
-					break;
-				default:
-					if (consonants > highest_consonants)
-						highest_consonants = consonants;
-					consonants = 0;
-					break;
+				consonants++;
+			}
+			else
+			{
+				if (consonants > highest_consonants)
+					highest_consonants = consonants;
+				consonants = 0;
 			}
 
 			/* Check vowels */
-			switch (*s)
+			if (strchr("aeiou", c))
 			{
-				case 'a':
-				case 'e':
-				case 'i':
-				case 'o':
-				case 'u':
-					vowels++;
-					break;
-				default:
-					if (vowels > highest_vowels)
-						highest_vowels = vowels;
-					vowels = 0;
-					break;
+				vowels++;
+			}
+			else
+			{
+				if (vowels > highest_vowels)
+					highest_vowels = vowels;
+				vowels = 0;
```

Comment:
It appears you are right, I benchmarked the `switch`-`case` vs `strchr()` over ~25 million runs, and there was an improvement when using the `switch`-`case`, but it appears to only be ~0.114 times as fast, and as such, I'm not sure how significant of a difference that would be to warrant a 20 line `switch`-`case`.